src/meili/map_matcher.cc

```diff
@@ -281,9 +281,12 @@ MatchResult FindMatchResult(const MapMatcher& mapmatcher,
       break;
     node_id = rbegin->nodeid();
     while (rbegin != rend && !prev_edge.Is_Valid()) {
-      // the very first state will have an initial dummy label with invalid edge/node ids
+      // the very first state will have an initial dummy label with invalid edge/node ids -
+      // except when the first state is a node, in which case the edge-id will be invalid
```

Comment:
well that is unexpected for sure! this is part of the contract that `meili/routing.cc` provides to the rest of the code. did you happen to have a look there to see why it doesnt put the initial dummy label in the labelset. specifically this method: https://github.com/valhalla/valhalla/blob/master/src/meili/routing.cc#L176 is supposed to do what the comment here (and also over there) says. very strange find indeed!

Reply:
I can have a look, but also see my comment about this being something the previous implementor thought to expect but didn't (author: Malakas).

Reply:
So I took a look at routing.cc, `set_origin`.  I also debugged it using my test case.  Based on the way the code is written, it seems very intentional that the origin state can either be 1) a dummy state (both node and edge-ids are invalid) or 2) a state with a valid node-id and invalid edge-id.  Maybe we can investigate this further when we spend some time on tech-debt, as I agree it would be much easier to understand and code against if the origin state was always a dummy state.